src/realm/dictionary.hpp

```diff
@@ -106,9 +106,9 @@ class Dictionary final : public CollectionBaseImpl<DictionaryBase>, public Colle
 
     Obj create_and_insert_linked_object(Mixed key);
 
-    DictionaryPtr insert_dictionary(StringData key);
+    void insert_dictionary(StringData key);
```

Comment:
Ok this is much better. However I think we need to review a bit how we construct a Collection in OS. I would like to be able to do something like:

```
auto t = group()->add_table();
auto col = t->add_column(type_Mixed, "Any");
auto obj = t->create_object();

List list{realm, obj, col}; //this tries to call obj.get_collection_ptr() but for a mixed that can have a collection stored, it is not ok

list.add_list(); //calls set_list()
auto l = list.insert_list(0); //calls insert_list(0); return get_list(0)
l.add(Mixed{5});
```